src/App.tsx

```diff
@@ -15,71 +18,75 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type AppState = {
+  isSelected: string
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, AppState> {
+  state: Readonly<AppState> = {
+    isSelected: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleClick = (name: string): void => {
+    if (this.state.isSelected === name) {
+      this.setState({ isSelected: '' });
+    } else {
+      this.setState({ isSelected: name });
+    }
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { isSelected } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {isSelected
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${isSelected} is selected`}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => this.handleClick('')}
+              />
+            </h1>
+          )
+          : <h1 className="title">No goods selected</h1>}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const btnClassName = cn({
+                button: true,
+                'is-info': good === isSelected,
+              });
```

Comment:
If the className should always be added you can write:
```suggestion
              const btnClassName = cn('button', {
                'is-info': good === isSelected,
              });
```